File: app/src/main/java/ivankuo/com/itbon2018/di/AppComponent.java
package ivankuo.com.itbon2018.di;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;
import ivankuo.com.itbon2018.GithubApp;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
        BuildersModule.class})
public interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(GithubApp application);
        AppComponent build();
    }

    void inject(GithubApp app);
}

File: app/src/main/java/ivankuo/com/itbon2018/di/AppComponent.kt
package ivankuo.com.itbon2018.di

import javax.inject.Singleton

import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import ivankuo.com.itbon2018.GithubApp

@Singleton
@Component(modules = arrayOf(
        AndroidSupportInjectionModule::class,
        AppModule::class,
        ActivityBuildersModule::class))
interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: GithubApp): Builder

        fun build(): AppComponent
    }

    fun inject(app: GithubApp)
}